src/main/java/org/folio/circulation/support/ItemRepository.java

```diff
@@ -361,4 +362,56 @@ private ItemRepository(
       .thenComposeAsync(this::fetchMaterialType)
       .thenComposeAsync(this::fetchLoanType);
   }
+
+  public <T extends ItemRelatedRecord> CompletableFuture<Result<MultipleRecords<T>>> fetchItemsForLoans(
+    Result<MultipleRecords<T>> result,
+    BiFunction<T, Item, T> includeItemMap) {
+
+    if (result.failed() || result.value().getRecords().isEmpty()) {
+      return CompletableFuture.completedFuture(result);
+    }
+
+    return result.combineAfter(r -> fetchItemsWithLoanTypes(getItemIds(r)),
+      (records, items) -> new MultipleRecords<>(
+        matchItemToRecord(records, items, includeItemMap),
+        records.getTotalRecords()));
+  }
+
+  private CompletableFuture<Result<Collection<Item>>> fetchItemsWithLoanTypes(
+    Collection<String> itemIds) {
+
+    return fetchFor(itemIds)
+      .thenComposeAsync(this::fetchLoanTypes);
```

Comment:
Why do the loan types need fetching for each item?

Reply:
It needs to fetch when loan type is present in circulation rule

Reply:
Why is the loan type itself needed? Only the ID is needed for the counting. Is this purely to provide an error message?

Reply:
Please check that every endpoint that uses this method has the appropriate permissions to fetch loan types and that the appropriate interface is included as a dependency.